spec/classes/config_spec.rb

```diff
@@ -1,47 +1,41 @@
 require 'spec_helper'
 describe 'caddy::config' do
-  context 'with default values for Redhat family release 6' do
-    let(:facts) do
-      {
-        os: {
-          family: 'RedHat',
-          name: 'RedHat',
-          release: { major: '6' },
-          architecture: 'x86_64'
-        }
-      }
-    end
+  on_supported_os.each do |os, os_facts|
+    svcpro =  case os_facts[:os]['release']['major']
+              when '6'
+                'redhat'
+              else
+                'systemd'
+              end
 
-    it { is_expected.to compile.with_all_deps }
-    it { is_expected.to contain_file('/etc/caddy') }
-    it { is_expected.to contain_file('/etc/caddy/Caddyfile') }
-    it { is_expected.to contain_file('/etc/caddy/config') }
-    it { is_expected.to contain_file('/var/log/caddy') }
-    it { is_expected.to contain_user('caddy') }
-    it { is_expected.to contain_group('caddy') }
-    it { is_expected.to contain_file('/etc/init.d/caddy') }
-  end
-  context 'with default values for Redhat family release 7' do
-    let(:facts) do
-      {
-        os: {
-          family: 'RedHat',
-          name: 'RedHat',
-          release: { major: '7' },
-          architecture: 'x86_64'
-        }
-      }
-    end
+    context "on #{os}" do
+      context 'with default values' do
+        let(:facts) { os_facts.merge(service_provider: svcpro) } # factordb doesn't provide service_provider
+
+        it { is_expected.to compile.with_all_deps }
```

Comment:
Please have a look at my latest pull request #37. I would like to extend the RSpec tests regarding the parameters.

Reply:
I rebased to pull your chages. If you were asking for more, I am not sure what you are asking.